process-context/c-and-cpp/example_ctx.c

```diff
@@ -0,0 +1,190 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License (Version 2.0).
+// This product includes software developed at Datadog (https://www.datadoghq.com/) Copyright 2025 Datadog, Inc.
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "otel_process_ctx.h"
+
+double burn_cpu(void) {
+  double x = 0.0;
+  for (int i = 0; i < 1000000; i++) {
+    x = (x+i)*(x+i);
+  }
+  return x;
+}
+
+void burn_cpu_forever(void) {
+  // avoid compiler optimization
+  volatile double x = 0.0;
+  while (true) {
+    x += burn_cpu();
+  }
+}
+
+void burn_cpu_for(int seconds) {
+  // avoid compiler optimization
+  volatile double x = 0.0;
+  time_t t0 = time(NULL);
+  while (time(NULL) - t0 < seconds) {
+    x += burn_cpu();
+  }
+}
+
+bool read_and_print_ctx(const char* prefix) {
+  otel_process_ctx_read_result result = otel_process_ctx_read();
+
+  if (!result.success) {
+    fprintf(stderr, "Failed to read context: %s\n", result.error_message);
+    return false;
+  }
+
+  printf(
+    "%s: service=%s, instance=%s, env=%s, version=%s, sdk=%s/%s/%s",
+    prefix,
```

Comment:
Nit: add PID here (will also be printed for the child) to make it easier to track updates with the dump script.

```suggestion
    "%s: pid=%d, service=%s, instance=%s, env=%s, version=%s, sdk=%s/%s/%s",
    prefix,
    getpid(),
```